### ISSUE TITLE
Change Log for NestJS Releases

### ISSUE THREAD

**Issue by miketeo** [nestjs/nest#6836]
## Feature Request
Maintain a change log for NestJS releases

## Is your feature request related to a problem? Please describe.
Currently, it is difficult to check what are the changes for each NestJS release.

## Describe the solution you'd like
A simple markdown file that details the changes for each NestJS release, and be committed to the git repository. The most recent changes should be inserted at the beginning of the file. The change log should at least describe the changes made for each major/minor release.

## Teachability, Documentation, Adoption, Migration Strategy
Not applicable.

## What is the motivation / use case for changing the behavior?
Maintaining a change log allows system maintainers to make an informed decision on whether to upgrade to the latest NestJS release. Security vulnerabilities or incompatible changes to the APIs will affect the priority of the upgrade.

**Comment by Tony133** [comment]
Hi @miketeo, 

There is a section in the documentation that explains the changes made see [here](https://docs.nestjs.com/migration-guide). And I think it's very useful.

While for a file `CHANGELOG.md` at the moment as you have seen it is not in the nest repository, but I don't know if it will be added. It would be extra work for the nest core team to manage both the documentation section and the CHANGELOG.md. So, if you put it in the Nest repository, in my opinion it should also be placed in all typical repositories (typeorm, mongoose, etc.). So extra work. (My personal opinion)
 
However, I always look at the versions / tags page like this:
https://github.com/nestjs/nest/releases/tag/v7.5.0. Here Kamil specifies everything that has been changed, improved, corrected or added.

**Comment by miketeo** [comment]
Yes, I saw the changes written up for v7.5.0, but there aren't any details for the subsequent releases after 7.5.0. 

I put up this request because running ```yarn outdated``` shows the latest release for @nestjs/core is at 7.6.15 which leads me to think that there should be some significant changes.

I just saw the NestJS team is still advertising 7.5.0 as the latest release. If we should follow the advertised "latest release" for building our web sites, then the current way of documenting changes is perfectly acceptable.
